CHANGELOG.md

```diff
@@ -1,3 +1,92 @@
+ * To better capture the context described above, `Event::PaymentFailed` has
+   been renamed to `Event::PaymentPathFailed` (#1084).
+ * A new event, `ChannelClosed` is provided by `ChannelManager` when a channel
+   is closed, including a reason and error message (if relevant, #997).
+ * `lightning-invoice` now considers invoices with sub-millisatoshi precision
+   to be invalid, and requires millisatoshi values during construction (thus
+   you must call `amount_milli_satoshis` instead of `amount_pico_btc`, #1057).
+ * The `BaseSign` interface now includes two new hooks which provide additional
+   information about commigment transaction signatures and revocation secrets
+   provided by our counterparty, allowing additional verification (#1039).
+ * The `BaseSign` interface now includes additional information for cooperative
+   close transactions, making it easier for a signer to verify requests (#1064).
+ * `Route` has two additional helper methods to get fees and amounts (#1063).
+ * `Txid` and `Transaction` objects can now be deserialized from responses when
+   using the HTTP client in the `lightning-block-sync` crate (#1037, #1061).
+
+## Bug Fixes
+ * Fix a panic when reading a lightning invoice with a non-recoverable
+   signature. Further, restrict lightning invoice parsing require payment
+   secrets and better handle a few edge cases as required by BOLT 11 (#1057).
+ * Fix a panic when receiving multiple messages (such as HTLC fulfill messages)
+   after a call to `chain::Watch::update_channel` returned
+   `Err(ChannelMonitorUpdateErr::TemporaryFailure)` with no
+   `ChannelManager::channel_monitor_updated` call in between (#1066).
+ * For multi-path payments, `Event::PaymentSent` is no longer generated
+   multiple times, once for each independent part (#1053).
+ * Multi-hop route hints in invoices are now considered in the default router
+   provided via `get_route` (#1040).
+ * The time peers have to respond to pings has been increased when building
+   with debug assertions enabled. This avoids peer disconnections on slow hosts
+   when running in debug mode (#1051).
+ * The timeout for the first byte of a response for requests from the
+   `lightning-block-sync` crate has been increased to 300 seconds to better
+   handle the long hangs in Bitcoin Core when it syncs to disk (#1090).
+
+## Serialization Compatibility
+ * 
+ * XXX There should at least be an entry for considerations around balances
+ * XXX There should at least be an entry for payment error graph updates
+ * XXX THere should at least be an entry for duplicate paymentfailed/sent for MPPs sent on old versions
+ * XXX probably more.
+
+In total, this release features 68 files changed, 5046 insertions, 1887
+deletions in 84 commits from 12 authors, in alphabetical order:
+ * Antoine Riard
+ * Devrandom
+ * Galder Zamarreño
+ * Giles Cope
+ * Jeffrey Czyz
+ * Joseph Goulden
+ * Matt Corallo
+ * Sergi Delgado Segura
+ * Tibo-lg
+ * Valentine Wallace
+ * abhik-99
```

Comment:
Abhik Banerjee

Reply:
Hmm, where is that? The git author is `abhik-99`, which I tend to prefer to keep unless the author expresses otherwise.

Reply:
Hmm, where is that? The git author is `abhik-99`, which I tend to prefer to keep unless the author expresses otherwise.

Reply:
Ah, I was looking at the corresponding GitHub page.

Reply:
Let's use git-authors as it would be simpler to automate.